src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -57,8 +57,25 @@ function normalizeDate(date) {
   return normalizedDate;
 }
 
+// recursive function to check a nested object for the 'toJSON' property
+function checkObjectForToJsonProperty(object) {
+  let objHasToJsonProperty = false;
+  Object.keys(object).forEach((key) => {
+    if (typeof object[key] === 'object') {
+      const hasToJsonProperty = checkObjectForToJsonProperty(object[key]);
```

Comment:
For cyclical references this will never stop running - probably OK for our use cases (the only good use case for objects is query objects and those are turned into JSON anyways which throws on cyclical references too)

Reply:
Hmm, how can I pass an object with cyclical references on the vega kibana functions?

Reply:
> For cyclical references this will never stop running

It will actually stop with "too much recursion", which sounds acceptable to me.

> Hmm, how can I pass an object with cyclical references on the vega kibana functions?

You can just pass something like `window` as an object argument:
```
{
  "events": {"type": "mousemove", "source": "window"},
  "update": "kibanaRemoveFilter(event.target.ownerDocument.defaultView)"
}
```